### PR TITLE
🧹  Clean up CMS javascript assets

### PR DESCRIPTION
**Pull request by josemigallas**
* Reduce/Remove dependency from jQuery to remove technical debt
* Remove mysterious functions from global scope
* Reduce boilerplate
* Remove old, broken and/or unused files and scripts
* Remove duplicated files and code
* Put files where they belong (dev portal ones under lib/developer_portal)

### CODE REVIEW

**Review by jlledom — COMMENTED** [review]
I don't feel very capable of reviewing this but I checked out the branch locally and it seems to work fine for me.

@josemigallas how are you sure these are not breaking changes?

File: app/views/layouts/_cms.html.erb [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+<%= javascript_include_tag 'vendor/jquery.tipsy.js' %>
+<%= javascript_include_tag 'vendor/jquery/extensions/jquery.cookie.js' %>
+<%= javascript_include_tag 'jquery.pjax.js' %>
+<%= javascript_include_tag 'provider/admin/cms/sidebar' %>
+
+<%= javascript_packs_with_chunks_tag 'codemirror', 'cms' %>
+<%= javascript_include_tag 'codemirror/modes/liquid' %>
+
+<%= javascript_tag do %>
+  <% timeout = (Rails.env.development? or Rails.env.test?) ? 10000 : 3000 -%>
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/views/layouts/_cms.html.erb:10]: Who knows, I decided to let it be (it comes from https://github.com/3scale/porta/blob/1ca3cb2e8f364a4e7f0f17c1bf4263ae36de21c4/app/views/provider/admin/cms/_includes.html.erb)

**Comment by mayorova** [comment]
Super hard to review this, as I don't understand some code well enough. But it seems that the features the changes affect work as expected.

**Comment by josemigallas** [comment]
@jlledom @mayorova assuming these changes are not breaking anything (I really think they would not), I suggest you go commit by commit and check nothing gets lost in translation. Each commit is self contained.

**Review by jlledom — APPROVED** [review]
I didn't review all commits one by one but I reviewed some of them and left some comments. Anyway, Everything looks right.

File: app/javascript/packs/cms.ts [hunk 2]
```diff
@@ -8,3 +8,38 @@ import 'jquery-ui/ui/widgets/tabs'
...
+ * selected section. When Path is focused and empty, it sets its value. This feature is weird and
+ * ineffective unless the input is focused.
+ *
+ * If CMS is loaded for the first time whithin a section, the input won't exist so it has to wait
+ * until DOMContentLoaded. On the other hand, if user is changing between sections the inputs are
+ * already in the DOM so partialPaths can be called right away.
+ *
+ * TODO: encapsulate this better in inputs/section_input.rb
+ */
+window.ThreeScale.partialPaths = (paths) => {
```
[the remark this answers is not in the document]
Comment (mayorova) [reply app/javascript/packs/cms.ts:94]: I think it's this one:

autocomplete

[the remark this answers is not in the document]
Comment (jlledom) [reply app/javascript/packs/cms.ts:94]: I see 👍

File: app/views/provider/admin/cms/_codemirror.html.erb [hunk 3]
```diff
@@ -16,6 +16,25 @@
     <% unless codemirror_disabled?  %>
       var codemirror = CodeMirror.fromTextArea(text_area[0], <%= json(opts).html_safe %>);
 
+      /**
+       * Prevents from exiting the CMS if there are unsaved changes. Alert message is set by browser.
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/views/provider/admin/cms/_codemirror.html.erb:21]: By other page do you mean another template? Or somewhere outside the CMS?

File: app/views/provider/admin/cms/_codemirror.html.erb [hunk 4]
```diff
@@ -76,6 +77,30 @@
                .on('cms.refresh', function(event){
                  codemirror.refresh();
                });
+
+      /**
+       * If it exists, set up a listener to "Preview" and "Preview draft" buttons that will give the
+       * user a chance to save any unsaved changes before opening the preview.
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/views/provider/admin/cms/_codemirror.html.erb:83]: If they click cancel, the preview should open *without saving* existing changes.

**Pull request merged**